src/include/distributed/multi_client_executor.h

```diff
@@ -72,6 +75,28 @@ typedef enum
 } BatchQueryStatus;
 
 
+/* Enumeration to track what the task associated with a connection is waiting for */
+typedef enum WaitStatus
```

Comment:
Looks good. I hesitated to put _just_ the word read or write in the last two values, since I thought the word _need_ or_wait_ would provide more information. But this is better than `WaitStatus`! I think it's better enough.